### PR TITLE
Fix footer link styles

### PR DESCRIPTION
**Pull request by FlyersPh9**
- Hyperlinks in the footer were inheriting a blue hover state.  This is no longer the case.
- Hyperlinks now have correct `:focus` styling.
- Added tests to check for the 2 fixes mentioned above.

### CODE REVIEW

File: src/footer/footer.scss [hunk 1]
```diff
@@ -44,8 +44,13 @@
 
       > a {
         @include iui-anchor;
-        @include themed {
-          color: t(iui-text-color-muted);
+        @include iui-focus;
+
+        &,
+        &:hover {
```
Comment (mayank99) [inline src/footer/footer.scss:50]: Um, why do we not want any hover state on the links?

Comment (FlyersPh9) [reply src/footer/footer.scss:50]: The hyperlinks do gain an underline on hover, but this is how they were styled prior to this bug.  I could introduce a `$iui-text-color-muted-overlay` similar to how we have in other places if you'd like to see a color change on hover as well.

Comment (mayank99) [reply src/footer/footer.scss:50]: Oh I was looking at the wrong image. Yeah I guess the underline is fine if that's how it was before.

File: backstop/scenarios/footer.js [hunk 2]
```diff
@@ -1,13 +1,32 @@
...
+  scenario('State hover', {
+    selectors: ['viewport'],
+    viewports: [{ width: 800, height: 600 }],
+    scrollToSelector: ['.iui-legal-footer'],
+    actions: [hover('#test-link')],
+  }),
+
+  // Focus states
+  //// Footer link
+  scenario('State focus', {
```
Comment (mayank99) [inline backstop/scenarios/footer.js:26]: Is it possible to have both focus and hover in the same test? 🤔

Comment (mayank99) [reply backstop/scenarios/footer.js:26]: Indeed it is. I think we should do that to reduce the number of test images.
image

Comment (bentleyvk) [reply backstop/scenarios/footer.js:26]: I think we thought about this before.

**Review by veekeys — DISMISSED** [review]

File: backstop/scenarios/footer.js (hunk 2 above)
Comment (FlyersPh9) [reply backstop/scenarios/footer.js]: Oooh, I love this suggestion.  I've pushed the update.

**Review by bentleyvk — APPROVED** [review]

**Review by mayank99 — APPROVED** [review]

**Pull request merged**